File: ios/Classes/Page/BaseAdPage.m
//
//  BaseAdPage.m
//  flutter_qq_ads
//
//  Created by zero on 2021/8/18.
//

#import "BaseAdPage.h"

@implementation BaseAdPage
// 添加广告事件
-(void) addAdEvent:(AdEvent *) event{
    if(self.eventSink!=nil){
        self.eventSink([event toMap]);
    }
}
// 显示广告
-(void)showAd:(NSString *)posId methodCall:(FlutterMethodCall *)call eventSink:(nonnull FlutterEventSink )events{
    self.posId=posId;
    self.eventSink=events;
    [self loadAd:call];
}
// 发送广告事件
- (void)sendEvent:(AdEvent *)event{
    if(self.eventSink!=nil){
        self.eventSink([event toMap]);
    }
}
// 发送广告事件
- (void)sendEventAction:(NSString *)action{
    AdEvent *event=[[AdEvent alloc] initWithAdId:self.posId andAction:action];
    [self sendEvent:event];
}
// 发送广告错误事件
- (void)sendErrorEvent:(NSInteger)errCode withErrMsg:(NSString *)errMsg{
    AdErrorEvent *event=[[AdErrorEvent alloc] initWithAdId:self.posId errCode:[NSNumber numberWithInteger:errCode]  errMsg:errMsg];
    [self sendEvent:event];
}

@end

File: android/src/main/java/com/zero/flutter_qq_ads/page/BaseAdPage.java
package com.zero.flutter_qq_ads.page;

import static com.zero.flutter_qq_ads.PluginDelegate.KEY_POSID;

import android.app.Activity;

import com.zero.flutter_qq_ads.event.AdErrorEvent;
import com.zero.flutter_qq_ads.event.AdEvent;
import com.zero.flutter_qq_ads.event.AdEventHandler;

import io.flutter.plugin.common.MethodCall;

/**
 * 基础广告页面
 */
public abstract class BaseAdPage {
    // 上下文
    protected Activity activity;
    // 广告位 id
    protected String posId;


    /**
     * 显示广告
     *
     * @param activity 上下文
     * @param call     方法调用
     */
    public void showAd(Activity activity, MethodCall call) {
        this.activity = activity;
        this.posId = call.argument(KEY_POSID);
        loadAd(call);
    }

    /**
     * 加载广告
     *
     * @param activity 上下文
     * @param call     方法调用
     */
    public abstract void loadAd(MethodCall call);


    /**
     * 发送广告事件
     *
     * @param event 广告事件
     */
    protected void sendEvent(AdEvent event) {
        AdEventHandler.getInstance().sendEvent(event);
    }

    /**
     * 发送广告事件
     *
     * @param action 操作
     */
    protected void sendEvent(String action) {
        sendEvent(new AdEvent(posId, action));
    }

    /**
     * 发送错误事件
     *
     * @param errCode 错误码
     * @param errMsg  错误事件
     */
    protected void sendErrorEvent(int errCode, String errMsg) {
        sendEvent(new AdErrorEvent(posId, errCode, errMsg));
    }
}
